### PR TITLE
Fix misleading error message when S3 credentials are wrong.

### LINKED ISSUE #291: Wrong S3 credentials results in misleading error message.
```
Command failed: Another error occured. `Failed to check for existence of index file.`. Cause: `StorageError(kind=InternalError, source=Request ID: None Body: )`
```

### PR DESCRIPTION
**Pull request by mosuka**
### Context / purpose

https://github.com/quickwit-inc/quickwit/issues/291

### Description
If the S3 credentials are wrong, a misleading error message will be displayed.

### How was this PR tested?
After setting the wrong credentials, run the following command:
```
cargo run -- new --index-config-path=./examples/index_configs/wikipedia_index_config.json --index-uri=s3://quickwit-dev/wikipedia-idx
```

The following error message will be displayed:
```
Command failed: Access forbidden. `The request credentials do not allow for this operation.`
```

### Checklist
- [x] tested locally

Closes #291

### CODE REVIEW

File: quickwit-storage/Cargo.toml [hunk 1]
```diff
@@ -24,6 +24,8 @@ thiserror = '1'
 rand = '0.8'
 lru = "0.6"
 serde = { version = "1.0", features = ["derive"] }
+# hyper = { version = "0.14", features = ["client", "http1", "http2", "tcp"] }
+hyper = "0.14"
```
Comment (guilload) [inline quickwit-storage/Cargo.toml:28]: Isn't adding `hyper` as a dependency for importing the constants of two widely known HTTP status codes a bit overkill?

File: quickwit-metastore/src/metastore/single_file_metastore.rs [hunk 2]
```diff
@@ -394,12 +421,14 @@ mod tests {
...
-    use crate::{Metastore, SingleFileMetastore, SplitMetadata, SplitState};
     use chrono::Utc;
+    use tokio::time::sleep;
+
     use quickwit_index_config::AllFlattenIndexConfig;
     use quickwit_storage::{MockStorage, StorageErrorKind};
-    use tokio::time::sleep;
+
+    use crate::{IndexMetadata, MetastoreError};
+    use crate::{Metastore, SingleFileMetastore, SplitMetadata, SplitState};
```
Comment (guilload) [inline quickwit-metastore/src/metastore/single_file_metastore.rs:431]: Just curious, do you use an automated tool to rearrange the imports?

Comment (mosuka) [reply quickwit-metastore/src/metastore/single_file_metastore.rs:431]: Basically, I used the cargo fmt. After that, I sorted them by standard packages, third party packages, Quickwit packages, and the packages itself.
If you prefer to just do `cargo fmt`, I will do that.

File: quickwit-storage/Cargo.toml (hunk 1 above)
Comment (mosuka) [reply quickwit-storage/Cargo.toml:28]: Okay, I’ll remove `hyper`.

File: quickwit-metastore/src/metastore/single_file_metastore.rs (hunk 2 above)
Comment (guilload) [reply quickwit-metastore/src/metastore/single_file_metastore.rs:431]: No, I like this ordering. We can't just expect people to follow the same convention without an automated way of doing so.

**Comment by fmassot** [comment]
@mosuka I made very small changes to improve wording :)

**Review by fmassot — APPROVED** [review]

**Comment by mosuka** [comment]
@fmassot Thanks!

**Pull request merged**